src/Api/DataProvider/TpayChannelItemDataProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Api\DataProvider;
+
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use CommerceWeavers\SyliusTpayPlugin\Api\Resource\TpayChannel;
+use CommerceWeavers\SyliusTpayPlugin\Tpay\Provider\TpayApiBankListProviderInterface;
+
+final class TpayChannelItemDataProvider implements ItemDataProviderInterface, RestrictedDataProviderInterface
+{
+    public function __construct(
+        private readonly TpayApiBankListProviderInterface $apiBankListProvider,
```

Comment:
AFAIR it's no longer a bank list provider, but the tpay channels provider.

Reply:
It is a bank list provider for UI part of plugin. It is first getting all the channels but the return of it, is a list of channels filtered by some fields which will point that the channel is a bank.